### ISSUE TITLE
Run ApiSix Helm chart without etcd Dependency

### ISSUE THREAD

**Issue by felpasl** [apache/apisix-helm-chart#686]
Like mention on this article:

https://apisix.apache.org/blog/2023/10/18/ingress-apisix/

change the ingress-controller deployment to give an option to deploy a lightweight instalation of the apisix based on this file: 

https://github.com/apache/apisix-ingress-controller/blob/master/samples/deploy/composite.yaml

**Comment by Friz-zy** [comment]
As I can see [here](https://github.com/apache/apisix-helm-chart/blob/49a65fe8bf0d1ff9124eab8efd364e94a03b8f7b/charts/apisix-ingress-controller/templates/deployment.yaml#L131) and [here](https://github.com/apache/apisix-helm-chart/blob/49a65fe8bf0d1ff9124eab8efd364e94a03b8f7b/charts/apisix-ingress-controller/templates/apisix-configmap.yaml#L37), currently this chart doesn't deploy separate etcd cluster, just start apisix container and use some local proxy (I think). However, I'm not a developer or maintainer though, just looking for a way to run a controller without etcd too :)

**Comment by Revolyssup** [comment]
@felpasl This has been added here https://github.com/apache/apisix-helm-chart/pull/671/files

**Comment by yuvalgtrustmi** [comment]
any update on that topic?
I would like to use the charts in order to deploy the "composite" deployment - how should I do that?

thanks!

**Comment by 4FunAndProfit** [comment]
Seems redundant with https://github.com/apache/apisix-helm-chart/issues/617 

@tao12345666333 , do you have an idea when the composite deployment will be available in chart? :) 

Thanks a lot

**Comment by rknightion** [comment]
@4FunAndProfit is it not already available via https://github.com/apache/apisix-helm-chart/pull/671/files# ?

**Comment by 4FunAndProfit** [comment]
@rknightion no, in composite mode, the ingress controller embeds the gateway container in the same pod

**Comment by vortegatorres** [comment]
I share these steps to run APISIX using the APISIX Helm chart version 2.10.0, in case it helps others: https://github.com/apache/apisix-ingress-controller/issues/2130#issuecomment-2900493642
